src/radical/pilot/messages.py

```diff
@@ -0,0 +1,76 @@
+
+
+from typing import Optional, Dict, Tuple, Any
+
+import radical.utils as ru
+
+
+# ------------------------------------------------------------------------------
+#
+class HeartbeatMessage(ru.Message):
+
+
+    _schema   = {'uid'      : str  }
+    _defaults = {'_msg_type': 'heartbeat',
```

Comment:
just `msg_type`? since that's what `ru.Message` has

p.s. and the same for `RPC*Message`

Reply:
I don't want to expose the type as part of the (app accessible) message, as the type hint is only used internally.  In other words, I don't want the user to be able (or tempted) to change that field.

Reply:
(1) not clear what do you mean with exposing, since it is in the schema
(2) the base class `ru.Message` has attribute `msg_type`